### PR TITLE
Makes a password validation on Android wallet on a common place with desktop and iOS

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/19859

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by AlexeyBarabash — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java [hunk 1]
```diff
@@ -65,38 +65,48 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
-                                    Toast.makeText(getActivity(), R.string.account_recovery_failed,
-                                                 Toast.LENGTH_SHORT)
-                                            .show();
-                                }
-                            });
+            EditText passwordEdittext = view.findViewById(R.id.restore_wallet_password);
+            String passwordInput = passwordEdittext.getText().toString();
+
+            KeyringController keyringController = getKeyringController();
+            assert keyringController != null;
```
Comment (wchen342) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java:74]: Why it is `assert` here? Won't that cause a crash if `KeyringController` is not ready for some reason?

**Comment by AlexeyBarabash** [comment]
Native code commit  for reference https://github.com/brave/brave-core/commit/c57729a43cb8d84dfb41a2566e34480691e7460a#diff-b44d65fd43bf18c3c2eba3fe2c9d553db219bb2a34b3762cfd8b990f8fe7aa68

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java (hunk 1 above)
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java:74]: it shouldn't happen. That fragment is called from `BraveWalletActivity`. All native stuff is loaded there. Everything should be up when we call that fragment.

Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java:74]: plus the keyringcontroller access is inside setOnClickListener

Comment (wchen342) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java:74]: Ah ok that makes sense.

**Pull request merged**